team/challenges/systems/worker.tex

```diff
@@ -0,0 +1,268 @@
+This is your work, and you are free to do whatever you feel is reasonable with it.
+In the scenario when you don't pass, you can open source it with any license and use it as a portfolio project.
+
+\subsection{Areas of focus}
+
+Teleport focuses on networking, infrastructure and security.
+
+These are the areas we will be evaluating in the submission:
+
+  \begin{itemize}
+  \item Use consistent coding style. We follow \href{https://github.com/golang/go/wiki/CodeReviewComments}{Golang Coding Style} for the Go language. If you are going to use a different language, please pick coding style guidelines and let us know what they are.
+  \item Create one test for authentication and another for the networking component.
+  \item Create one test for unhappy scenario.
+  \item Make sure builds are reproducible. Pick any vendoring/packaging system that will allow us to get consistent build results.
+  \item Ensure error handling and error reporting is consistent. The system should report clear errors and not crash under non-critical conditions.
+  \item Avoid concurrency and networking errors. Most of the issues we've seen in production are related to data races, networking error handling or goroutine leaks. We will be looking for those errors in your code.
+  \item Security. Use strong authentication and simplest, but robust authorization. Set up the strongest transport encryption you can. Test it.
+  \end{itemize}
+
+\subsection{Trade-offs}
+       
+Write as little code as possible, otherwise this task will consume too much time and code quality will suffer.
+
+Please cut corners, for example configuration tends to take a lot of time, and is not important for this task.
+
+Use hard codes as much as possible and simply add TODO items showing your thinking, for example:
+
+\begin{lstlisting}[caption=TODO example]
+  
+  // TODO: Add configuration system.
+  // Consider using CLI library to support both
+  // environment variables and reasonable default values,
+  // for example https://github.com/alecthomas/kingpin
+
+\end{lstlisting}
+
+Comments like this one are really helpful to us.
+They save yourself a lot of time and demonstrate that you've spent time thinking about this problem and provide a clear path to solution.
+
+Consider making other reasonable trade-offs. Make sure you communicate them to the interview team.
+
+Here are some other trade-offs that will help you to spend less time on the task:
+
+
+Do not implement a system that scales or is highly performing. Describe which performance improvements would you add in the future?
+High availability. It is OK if the system is not highly available. Write down how you would make the system highly available and why your system is not.
+Do not try to achieve full test coverage. This will take too long. Take two key components, e.g. authentication/authorization layer and networking and implement one-two test cases that demonstrate your approach to testing.
```

Comment:
```suggestion
Do not try to achieve full test coverage. This will take too long. Take two key components, e.g. authentication/authorization layer and networking and implement one or two test cases that demonstrate your approach to testing.
```